File: scripts/test_solutions/846_hand_of_straights.py
from collections import Counter

def isNStraightHand(hand, groupSize):
    if len(hand) % groupSize != 0:
        return False
    count = Counter(hand)
    for start in sorted(count.keys()):
        cnt = count[start]
        if cnt > 0:
            for i in range(start, start + groupSize):
                if count[i] < cnt:
                    return False
                count[i] -= cnt
    return True

File: scripts/test_solutions/846_hand_of_straights.js
var isNStraightHand = function(hand, groupSize) {
    if (hand.length % groupSize !== 0) return false;
    const count = new Map();
    for (const card of hand) count.set(card, (count.get(card) || 0) + 1);
    const sorted = [...count.keys()].sort((a, b) => a - b);
    for (const start of sorted) {
        const cnt = count.get(start);
        if (cnt > 0) {
            for (let i = start; i < start + groupSize; i++) {
                if ((count.get(i) || 0) < cnt) return false;
                count.set(i, count.get(i) - cnt);
            }
        }
    }
    return true;
};
